api/python/t4/packages.py

```diff
@@ -187,28 +187,49 @@ def get(self):
         """
         return _to_singleton(self.physical_keys)
 
-    def deserialize(self):
+    def deserialize(self, deserializer=None):
         """
         Returns the object this entry corresponds to.
 
+        Custom Deserializers:
+            If `deserializer` is given, it is passed the physical key
+            of the data that would have been deserialized, then its
+            return value is returned.
+
+            Example for decoding a json blob:
+                >>> des = lambda k: json.loads(Path(urlparse(k).path).read_text())
+                >>> obj = p[k](des)  # obj is the deserialized JSON blob
```

Comment:
I know I brought this up in Quilt2... but this is the same thing as:
```
obj = des(pk[k].get())
```
In fact, at that point, no need to create a lambda - just call:
```
obj = json.loads(...(p[k].get())
```
So I'm still not convinced that passing a deserializer is actually useful.

Though I could imagine it being useful if e.g. we were passing an open file descriptor into `des`: then we'd handle opening/closing the file, and the user wouldn't have to worry about that.